### PR TITLE
Added OfficeMeta Service

### PR DESCRIPTION
**Pull request by ms-xy**
Requires 7832ab6b0d4d097c4aa3c309d6fc6335236cd0bb from the holmes-totem-library to be merged first. See HolmesProcessing/Holmes-Totem-Service-Library#2

### CODE REVIEW

File: config/totem.conf.example [hunk 1]
```diff
@@ -50,6 +50,10 @@ totem {
       uri = ["http://127.0.0.1:7703/metadata/", "http://127.0.0.1:7707/metadata/"]
       resultRoutingKey = "metadata.result.static.totem"
     }
+    office_meta {
+      uri = ["http://127.0.0.1:7730/office_meta/"]
+      resultRoutingKey = "office_meta.result.static.totem"
```
Comment (webstergd) [inline config/totem.conf.example]: Please change office_meta and office_meta.result.static.totem to officemeta so the styles match.

File: src/main/scala/org/novetta/zoo/driver/driver.scala [hunk 2]
```diff
@@ -4,6 +4,7 @@ import java.util.concurrent.{Executors, ExecutorService}
 
 import akka.actor.{ActorRef, ActorSystem, Props}
 import org.novetta.zoo.actors._
+import org.novetta.zoo.services.office_meta.{OfficeMetaSuccess, OfficeMetaWork}
```
Comment (webstergd) [inline src/main/scala/org/novetta/zoo/driver/driver.scala]: same here. Just officemeta

File: src/main/scala/org/novetta/zoo/driver/driver.scala [hunk 3]
```diff
@@ -126,9 +130,10 @@ object driver extends App with Instrumented {
 
     def workRoutingKey(work: WorkResult): String = {
       work match {
+        case x: MetadataSuccess => conf.getString("totem.enrichers.metadata.resultRoutingKey")
```
Comment (webstergd) [inline src/main/scala/org/novetta/zoo/driver/driver.scala:134]: Thank you for making it alphabetical

File: src/main/scala/org/novetta/zoo/services/office_meta/OfficeMetaREST.scala [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package org.novetta.zoo.services.zipmeta
```
Comment (webstergd) [inline src/main/scala/org/novetta/zoo/services/office_meta/OfficeMetaREST.scala]: please fix the typo for the package and default routing keys.

File: src/main/scala/org/novetta/zoo/services/office_meta/service.conf.example [hunk 5]
```diff
@@ -0,0 +1,11 @@
+[metadata]
+Name        = OfficeMeta
+Version     = 1.0
+Description = ./README.md
+Copyright   = Copyright 2015 Holmes Processing
```
Comment (webstergd) [inline src/main/scala/org/novetta/zoo/services/office_meta/service.conf.example]: please change to 2016

File: src/main/scala/org/novetta/zoo/services/office_meta/service.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
+import tornado
```
Comment (webstergd) [inline src/main/scala/org/novetta/zoo/services/office_meta/service.py]: Please add comments for each import block. Look at peinfo for an example.

File: src/main/scala/org/novetta/zoo/services/office_meta/service.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+from office_meta      import OfficeParser, OfficeMetaError
+from library.services import ServiceResultSet, ServiceConfig
+
+import binascii
+
+
+# get config and init Tornado
+Config = ServiceConfig("./service.conf")
+define("port", default=int(Config.settings.port,10), help="port to run", type=int)
```
Comment (webstergd) [inline src/main/scala/org/novetta/zoo/services/office_meta/service.py]: please remove default port 10. Would rather have the setting match the docker file or ideally just fail.

**Comment by webstergd** [comment]
Thanks dude. Please test. Additionally I would like the following changes: 
Please fix the typos, adjust to match the naming convention of the other services, provide a meaningful commit description, and add a comment on why you selected to use the CRITs' officemeta service. On your justification I want to know why you believe porting the CRITs' Service was a better verses the other available options or writing your own.

**Comment by ms-xy** [comment]
All changes as requested, sbt assembly runs through, sorry for the mistakes on the naming convention.
You might want to look at the cleanup I've done on the imports, maybe we should do that for other services using tornado as well.

**Comment by ms-xy** [comment]
I'll make a writeup as requested tomorrow.

**Comment by ms-xy** [comment]
As requested a short writeup as to why the CRITs implementation was chosen:
CRITs is state-of-the-art, widely accepted, their services are working and thus a reliable source.
Since we try to implement as much compatibility with CRITs as possible, it's good to start out with their implementation and incrementally improve it. (Implement support for newer office document types for example)

**Comment by ms-xy** [comment]
Extended it to now parse Office-XML (2003 version) as well as Office Open XML (Microsoft Office XML 2007) formatted files.
A new problem I see about this service now is that the return format can't be 100% conform with the previous format, as that one relied on the document structure of Microsofts OLE files.
I tried to keep it as close as possible for the new file types, but it isn't quite the same. :-1:

**Comment by webstergd** [comment]
what's the current status of this? are you still working it?

**Comment by ms-xy** [comment]
For some reason I thought it was merged. I'll have to revisit it to see what the errors are, I haven't touched it in 4 months ...
I'll get to it later this week, thursday or friday.

**Comment by ms-xy** [comment]
Pushing this one back a bit, as I'm currently still working on the PT service.

**Comment by webstergd** [comment]
closing pull request until the code is ready to be submitted. When you are ready, please resubmit the pull request.

**Pull request closed without merging**